### PR TITLE
Saas 12.2 elearning fiximp dbe

### PR DESCRIPTION
**Pull request by dbeguin**

### CODE REVIEW

File: addons/website_slides/models/slide_channel.py [hunk 1]
```diff
@@ -372,25 +371,40 @@ def _action_add_members(self, target_partners, **member_values):
...
         return self.env['slide.channel.partner'].sudo()
 
+    def _filter_target_members(self, target_partners, **member_values):
+        filtered_target_partner_ids = []
+        public_users = self.env['res.users'].sudo()
+        public_groups = self.env.ref("base.group_public", raise_if_not_found=False)
+        if public_groups:
+            public_users = public_groups.sudo().with_context(active_test=False).mapped("users")
+        for target_partner in target_partners:
+            is_public = target_partner.sudo().with_context(active_test=False).user_ids in public_users if public_users else False
```
Comment (jem-odoo) [inline addons/website_slides/models/slide_channel.py]: Maybe you should check `_is_public()` on each user.

File: addons/website_profile/controllers/main.py [hunk 2]
```diff
@@ -210,25 +211,28 @@ def view_all_users_page(self, page=1, **searches):
...
-        # Get the Top 3 users
-        top3_users = User.sudo().search(dom, limit=3, order='karma DESC')
-        top3_user_values = [self._prepare_all_users_values(user, position+1) for position, user in enumerate(top3_users)]
+        # Searches : filter after search to get the correct position.
+        search_term = searches.get('search')
+        if search_term:
+            dom = expression.AND([['|', ('name', 'ilike', searches.get('search')), ('company_id.name', 'ilike', searches.get('search'))], dom])
+        elif page == 1:
+            # Get the Top 3 users
+            top3_users = User.sudo().search(dom, limit=3, order='karma DESC')
```
[the remark this answers is not in the document]
Comment (dbeguin) [reply addons/website_profile/controllers/main.py]: Indeed, as I finally took the decision to not show the podium in not first page I can now do that. Right !

File: addons/website_slides/models/slide_channel.py (hunk 1 above)
Comment (dbeguin) [reply addons/website_slides/models/slide_channel.py]: I think I was in a parallel world when I did this branch..

**Comment by dbeguin** [comment]
integrated into #31493 - pager on users and slides by category - and #31321- Email validation and profile imp -

**Pull request closed without merging**